app/services/curation_concerns/workflow/permission_query.rb

```diff
@@ -257,7 +257,10 @@ def scope_users_for_entity_and_roles(entity:, roles:)
             .and(entity_responsibilities[:entity_id].eq(entity.id))
         )
 
-        sub_query_for_user = agent_table.project(agent_table[:proxy_for_id]).where(
+        # PostgreSQL requires an explicit cast from string to integer
+        cast = Arel::Nodes::NamedFunction.new "CAST", [agent_table[:proxy_for_id].as("integer")]
```

Comment:
I believe the core issue here is that Sipity agent ids are strings, to allow for some magic ids such as "admin" or "registered".  But this query should only retrieve the integer ids that are linked to roles and works.  (If that's not true, then casting the user.id to string would be better).

I don't know if @jcoyne has more background on this from the original fix in Hyku.